deflect/server/Tile.h

```diff
@@ -0,0 +1,85 @@
+/* Redistribution and use in source and binary forms, with or        */
+/* without modification, are permitted provided that the following   */
+/* conditions are met:                                               */
+/*                                                                   */
+/*   1. Redistributions of source code must retain the above         */
+/*      copyright notice, this list of conditions and the following  */
+/*      disclaimer.                                                  */
+/*                                                                   */
+/*   2. Redistributions in binary form must reproduce the above      */
+/*      copyright notice, this list of conditions and the following  */
+/*      disclaimer in the documentation and/or other materials       */
+/*      provided with the distribution.                              */
+/*                                                                   */
+/*    THIS  SOFTWARE IS PROVIDED  BY THE  UNIVERSITY OF  TEXAS AT    */
+/*    AUSTIN  ``AS IS''  AND ANY  EXPRESS OR  IMPLIED WARRANTIES,    */
+/*    INCLUDING, BUT  NOT LIMITED  TO, THE IMPLIED  WARRANTIES OF    */
+/*    MERCHANTABILITY  AND FITNESS FOR  A PARTICULAR  PURPOSE ARE    */
+/*    DISCLAIMED.  IN  NO EVENT SHALL THE UNIVERSITY  OF TEXAS AT    */
+/*    AUSTIN OR CONTRIBUTORS BE  LIABLE FOR ANY DIRECT, INDIRECT,    */
+/*    INCIDENTAL,  SPECIAL, EXEMPLARY,  OR  CONSEQUENTIAL DAMAGES    */
+/*    (INCLUDING, BUT  NOT LIMITED TO,  PROCUREMENT OF SUBSTITUTE    */
+/*    GOODS  OR  SERVICES; LOSS  OF  USE,  DATA,  OR PROFITS;  OR    */
+/*    BUSINESS INTERRUPTION) HOWEVER CAUSED  AND ON ANY THEORY OF    */
+/*    LIABILITY, WHETHER  IN CONTRACT, STRICT  LIABILITY, OR TORT    */
+/*    (INCLUDING NEGLIGENCE OR OTHERWISE)  ARISING IN ANY WAY OUT    */
+/*    OF  THE  USE OF  THIS  SOFTWARE,  EVEN  IF ADVISED  OF  THE    */
+/*    POSSIBILITY OF SUCH DAMAGE.                                    */
+/*                                                                   */
+/* The views and conclusions contained in the software and           */
+/* documentation are those of the authors and should not be          */
+/* interpreted as representing official policies, either expressed   */
+/* or implied, of The University of Texas at Austin.                 */
+/*********************************************************************/
+
+#ifndef DEFLECT_SERVER_TILE_H
+#define DEFLECT_SERVER_TILE_H
+
+#include <deflect/types.h>
+
+#include <QByteArray>
+
+namespace deflect
+{
+namespace server
+{
+/**
+ * A Tile is a sub-region of an image in a Frame.
+ */
+struct Tile
+{
+    /** @name Coordinates */
+    //@{
+    uint32_t x = 0u; /**< The x position in pixels. */
+    uint32_t y = 0u; /**< The y position in pixels. */
+    //@}
+
+    /** @name Dimensions */
+    //@{
+    uint32_t width = 0u;  /**< The width in pixels. */
+    uint32_t height = 0u; /**< The height in pixels. */
+    //@}
```

Comment:
those four and the data type is a repetition from the segment parameters. reuse them here?

Reply:
This is a tricky point. The existing structs Segment and SegementParameters are shared between Deflect and DeflectServer because they are part of the networking protocol. However, they were also part of the public DeflectServer API because they were exposed through the Frame object. The problem was that DeflectServer depends on Deflect, so the shared classes should logically be in Deflect, but the headers have no reason to be public in Deflect because they are not used by client code. In the end I decided to keep the legacy structs private in Deflect and introduce this new Tile struct for use in Frame in DeflectServer that combines the two. It's nicer to use without the nested SegementParameters object and the public API can evolve separately from the networking API, at the cost of some duplication.